### PR TITLE
[SPARK-47517][CORE][SQL] Prefer Utils.bytesToString for size display

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

I found almost everywhere in the codebase uses Spark's built-in `Utils.bytesToString` to convert bytes to human-readable string, but only a few places use `org.apache.commons.io.FileUtils.byteCountToDisplaySize`.

### Why are the changes needed?

Minor code improvement, for display consistency, and reduce the couple with 3-rd libs.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The change only affects debug logs and benchmark result files.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/MapStatusesSerDeserBenchmark.scala [hunk 1]
```diff
@@ -85,12 +86,11 @@ object MapStatusesSerDeserBenchmark extends BenchmarkBase {
     }
 
     benchmark.run()
-    // scalastyle:off
-    import org.apache.commons.io.FileUtils
+    // scalastyle:off println
     benchmark.out.println("Compressed Serialized MapStatus sizes: " +
-      FileUtils.byteCountToDisplaySize(serializedMapStatusSizes))
```
Comment (yaooqinn) [inline core/src/test/scala/org/apache/spark/MapStatusesSerDeserBenchmark.scala:91]: shall we update the benchmark golden files?

Comment (pan3793) [reply core/src/test/scala/org/apache/spark/MapStatusesSerDeserBenchmark.scala:91]: I'm doing that.

Comment (yaooqinn) [reply core/src/test/scala/org/apache/spark/MapStatusesSerDeserBenchmark.scala:91]: Thank you, and excuse me for being a bit pushy.

Comment (pan3793) [reply core/src/test/scala/org/apache/spark/MapStatusesSerDeserBenchmark.scala:91]: The corresponding benchmark results are updated

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master.

Thank you @pan3793

**Review by dongjoon-hyun — APPROVED** [review]
To @pan3793 , could you try to prevent `FileUtils.byteCountToDisplaySize` completely by enforcing in scalastyle? I guess you can follow the following style.

https://github.com/apache/spark/blob/32dfdd305aec955ef25ba6853e28e99e52dfb48b/scalastyle-config.xml#L463-L468

**Comment by dongjoon-hyun** [comment]
Oh, never mind. Let me do that separately. Thank you all.

**Comment by dongjoon-hyun** [comment]
Here is the follow-up.
- #45670

**Comment by pan3793** [comment]
@dongjoon-hyun thank you for making a follow-up.

**Pull request closed without merging**